ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/MainnetTransactionValidator.java

```diff
@@ -109,6 +116,10 @@ public MainnetTransactionValidator(
       return signatureResult;
     }
 
+    if (transaction.getChainId().isEmpty() && !acceptUnprotectedTransactions) {
```

Comment:
We can only enforce this if there's actually a `chainId` specified.  I think this should be:
```suggestion
    if (!acceptUnprotectedTransactions && chainId.isPresent() && transaction.getChainId().isEmpty()) {
```

We should also make sure we have tests for all possible permutations here (with / without chainId, with / without unprotected txs allowed, txs with / without a chainId).

Reply:
I missed that! Thanks for pointing it out. I will add tests for it.